builder/main.lisp

```diff
@@ -0,0 +1,81 @@
+;;; Core
+
+(defun main ()
+  (with-open-file (in-stream "template.md" :if-does-not-exist :error)
+    (let ((template-string (make-string (file-length in-stream))))
+      (read-sequence template-string in-stream)
+      (with-open-file (out-stream "../README.md" :direction :output :if-does-not-exist :create :if-exists :overwrite)
+        (write-string (str-replace template-string "{{table}}" (generate-table (get-cats-names))) out-stream)))))
+        
+(defun get-cats-names ()
+  "Return a list of the cats names."
+  (let* ((directory (uiop:subdirectories "../cats"))
+         (folders (map 'list #'get-folder-name directory)))
+    folders))
+
+(defun get-cat-pictures (cat-name)
+  (remove-if #'is-not-image (uiop:directory-files (str-replace "../cats/$dir/" "$dir" cat-name))))
+
+(defun get-cat-profile-picture (cat-name)
+  (let* ((files (get-cat-pictures cat-name))
+         (first-file (file-namestring (first files))))
+    (str-replace (str-replace "./cats/$name/$file" "$name" cat-name) "$file" first-file)))
+
+(defun get-folder-name (pathname)
+  "Return only the pathname's folder as string instead of the entire path."
+  (first (last (pathname-directory pathname))))
+
+;; Generators
+
+(defun generate-table (cats-names)
+  (str-replace "<table>$1</table>" "$1" (generate-table-rows cats-names)))
+
+(defun generate-table-cell (cat-name)
+  (str-replace (str-replace (str-replace "<td align=\"center\"><a href=\"./cats/$3\">$1<strong>$2</strong></a></td>" "$1" (generate-img (get-cat-profile-picture cat-name))) "$2" cat-name) "$3" cat-name))
+
+(defun generate-table-rows (cats-names)
```

Comment:
You are using later a cdddr but the input is not clear. Give a example of cats-names structure on the defun docstring

Reply:
I removed the `cdddr` and inserted a docstring. Is it clear now?

Reply:
Yes, it is.